contracts/Marketplace.sol

```diff
@@ -92,6 +98,61 @@ contract Marketplace is Collateral, Proofs {
     require(token.transfer(slot.host, amount), "Payment failed");
   }
 
+  /// @notice Withdraws storage request funds back to the client that deposited them.
+  /// @dev Request must be expired, must be in RequestState.New, and the transaction must originate from the depositer address.
+  /// @param requestId the id of the request
+  function withdrawFunds(bytes32 requestId) public marketplaceInvariant {
+    Request memory request = requests[requestId];
+    require(block.timestamp > request.expiry, "Request not yet timed out");
+    require(request.client == msg.sender, "Invalid client address");
+    RequestContext storage context = requestContexts[requestId];
+    require(context.state == RequestState.New, "Invalid state");
+
+    uint256 amount = _price(request);
```

Comment:
This is ok for now, but will have to be changed once we start paying out hosts for the time they fill a slot. The amount that we paid to hosts will then have to be deducted from the price.

Reply:
Added a TODO